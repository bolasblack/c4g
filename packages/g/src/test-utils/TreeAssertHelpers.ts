import {
  UnitTestTree,
  SchematicTestRunner,
} from '@angular-devkit/schematics/testing'

export function matchFileStructureSnapshot(
  creator: () => Promise<{ tree: UnitTestTree }>,
) {
  it('match file structure snapshot', async () => {
    const { tree } = await creator()
    expect(tree.files).toMatchSnapshot('file structure')
  })
}

export function matchFileContentSnapshot(
  creator: () => Promise<{ tree: UnitTestTree }>,
) {
  it('match file content snapshot', async () => {
    const { tree } = await creator()
    tree.files.forEach(file => {
      expect(tree.readContent(file)).toMatchSnapshot(`file content: ${file}`)
    })
  })
}

export function matchTasksSnapshot(
  creator: () => Promise<{ runner: SchematicTestRunner }>,
) {
  it('match tasks snapshot', async () => {
    const { runner } = await creator()
    expect(runner.tasks).toMatchSnapshot('rule tasks')
  })
}

export function matchRunResult(
  creator: () => Promise<{
    tree: UnitTestTree
    runner: SchematicTestRunner
  }>,
) {
  matchFileStructureSnapshot(creator)
  matchFileContentSnapshot(creator)
  matchTasksSnapshot(creator)
}

export function assertTreeSnapshot(
  runner: SchematicTestRunner,
  tree: UnitTestTree,
) {
  expect(tree.files).toMatchSnapshot('file structure')

  tree.files.forEach(file => {
    expect(tree.readContent(file)).toMatchSnapshot(`file content: ${file}`)
  })

  expect(runner.tasks).toMatchSnapshot('rule tasks')
}
