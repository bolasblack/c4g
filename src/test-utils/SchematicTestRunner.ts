import { Observable, of as observableOf } from 'rxjs'
import { map } from 'rxjs/operators'
import { logging } from '@angular-devkit/core'
import {
  HostTree,
  Tree,
  Collection,
  TypedSchematicContext,
  ExecutionOptions,
  Schematic,
} from '@angular-devkit/schematics'
import {
  UnitTestTree,
  SchematicTestRunner as SchematicTestRunnerBase,
} from '@angular-devkit/schematics/testing'
import { NodeModulesTestEngineHost } from '@angular-devkit/schematics/tools'

export interface CallSchematic<SchematicSchemaT> {
  (
    schematic: Schematic<{}, {}>,
    options: SchematicSchemaT,
    tree: Observable<Tree>,
    parentContext: Partial<TypedSchematicContext<{}, {}>>,
  ): Observable<Tree>
}

export class SchematicTestRunner extends SchematicTestRunnerBase {
  createContext(
    schematicName: string,
    parent?: Partial<TypedSchematicContext<{}, {}>>,
    executionOptions?: Partial<ExecutionOptions>,
  ) {
    const _collection: Collection<{}, {}> = this['_collection']
    const _logger: logging.Logger = this['_logger']
    const { engine } = this

    const schematic = _collection.createSchematic(schematicName, true)

    return engine.createContext(
      schematic,
      {
        logger: _logger,
        ...parent,
      },
      executionOptions,
    )
  }

  runSchematicAsync<SchematicSchemaT>(
    schematicName: string,
    options?: SchematicSchemaT,
    tree?: Tree,
    callSchematic?: CallSchematic<SchematicSchemaT>,
  ): Observable<UnitTestTree> {
    const _collection: Collection<{}, {}> = this['_collection']
    const _engineHost: NodeModulesTestEngineHost = this['_engineHost']
    const _logger: logging.Logger = this['_logger']

    const schematic = _collection.createSchematic(schematicName, true)
    const host = observableOf(tree || new HostTree())
    _engineHost.clearTasks()

    callSchematic = callSchematic || this._callSchematic.bind(this)

    return callSchematic!(schematic, options as any, host, {
      logger: _logger,
    }).pipe(map(tree => new UnitTestTree(tree)))
  }

  runSchematic<SchematicSchemaT extends {}>(
    schematicName: string,
    options?: SchematicSchemaT,
    tree?: Tree,
    callSchematic?: CallSchematic<SchematicSchemaT>,
  ): UnitTestTree {
    const _collection: Collection<{}, {}> = this['_collection']
    const _engineHost: NodeModulesTestEngineHost = this['_engineHost']
    const _logger: logging.Logger = this['_logger']

    const schematic = _collection.createSchematic(schematicName, true)

    let result: UnitTestTree | null = null
    let error
    const host = observableOf(tree || new HostTree())
    _engineHost.clearTasks()

    callSchematic = callSchematic || this._callSchematic.bind(this)

    callSchematic!(schematic, options as any, host, {
      logger: _logger,
    }).subscribe(t => (result = new UnitTestTree(t)), e => (error = e))

    if (error) {
      throw error
    }

    if (result === null) {
      throw new Error('Schematic is async, please use runSchematicAsync')
    }

    return result
  }

  protected _callSchematic<SchematicSchemaT extends {}>(
    schematic: Schematic<{}, {}>,
    options: SchematicSchemaT | undefined,
    tree: Observable<Tree>,
    parentContext: Partial<TypedSchematicContext<{}, {}>>,
  ) {
    return schematic.call(options || {}, tree, parentContext)
  }
}
