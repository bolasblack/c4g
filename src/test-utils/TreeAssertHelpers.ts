import { UnitTestTree } from '@angular-devkit/schematics/testing'
import { SchematicTestRunner } from '../test-utils/SchematicTestRunner'

export function assertTreeSnapshot(
  runner: SchematicTestRunner,
  tree: UnitTestTree,
) {
  // @ts-ignore
  expect(tree.files).toMatchSnapshot('file structure')
  tree.files.forEach(file => {
    // @ts-ignore
    expect(tree.readContent(file)).toMatchSnapshot(`file content: ${file}`)
  })

  // @ts-ignore
  expect(runner.tasks).toMatchSnapshot('rule tasks')
}
