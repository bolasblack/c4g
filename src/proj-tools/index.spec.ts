import * as path from 'path'
import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { assertTreeSnapshot } from '../test-utils/TreeAssertHelpers'
import { Options, IncludeItem } from './index'

const collectionPath = path.join(__dirname, '../collection.json')

const toolNames = Object.keys(IncludeItem)

describe('proj-tools', () => {
  const test = (options: Options) => {
    const runner = new SchematicTestRunner('schematics', collectionPath)
    const resultTree = runner.runSchematic(
      'proj-tools',
      { ...options, interactive: false },
      Tree.empty(),
    )
    assertTreeSnapshot(runner, resultTree)
  }

  describe('with option `include`', () => {
    it('works with empty value', () => {
      test({})
    })

    toolNames.forEach(toolName => {
      it(`works with \`IncludeItem.${toolName}\``, () => {
        test({ include: [IncludeItem[toolName]] })
      })
    })
  })

  describe('with option `jestReact`', () => {
    it('works', () => {
      test({ include: [IncludeItem.Jest], interactive: false, jestReact: true })
    })

    it('proxy `cwd` option', () => {
      test({ include: [IncludeItem.Jest], interactive: false, cwd: 'test' })
    })

    it('do nothing if `jest` not included', () => {
      test({ include: [], interactive: false, jestReact: true })
    })
  })
})
