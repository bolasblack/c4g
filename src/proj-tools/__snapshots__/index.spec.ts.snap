// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proj-tools with option \`include\` with \`IncludeItem.Commitlint\` match file content snapshot: file content: /.commitlintrc.js 1`] = `
"module.exports = {
  extends: ['@c4605/toolconfs/commitlintrc'],
  rules: {},
}
"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Commitlint\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"commitlint\\": \\"^1.0.0\\",
    \\"@commitlint/config-conventional\\": \\"^1.0.0\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Commitlint\` match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
  "/.commitlintrc.js",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Commitlint\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` in common case match file content snapshot: file content: /.eslintrc.json 1`] = `
"{
  \\"extends\\": [
    \\"./node_modules/@c4605/toolconfs/eslintrc.base\\",
    \\"./node_modules/@c4605/toolconfs/eslintrc.ts\\"
  ],
  \\"parserOptions\\": {
    \\"project\\": \\"./tsconfig.json\\"
  },
  \\"rules\\": {}
}
"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` in common case match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"eslint\\": \\"^1.0.0\\",
    \\"@typescript-eslint/parser\\": \\"^1.0.0\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"^1.0.0\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` in common case match file structure snapshot: file structure 1`] = `
Array [
  "/.eslintrc.json",
  "/package.json",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` in common case match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` install eslint-plugin-prettier when including prettier match file content snapshot: file content: /.eslintrc.json 1`] = `
"{
  \\"extends\\": [
    \\"./node_modules/@c4605/toolconfs/eslintrc.base\\",
    \\"./node_modules/@c4605/toolconfs/eslintrc.ts\\",
    \\"./node_modules/@c4605/toolconfs/eslintrc.prettier\\"
  ],
  \\"parserOptions\\": {
    \\"project\\": \\"./tsconfig.json\\"
  },
  \\"rules\\": {}
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` install eslint-plugin-prettier when including prettier match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"eslint\\": \\"^1.0.0\\",
    \\"eslint-config-prettier\\": \\"^1.0.0\\",
    \\"@typescript-eslint/parser\\": \\"^1.0.0\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"^1.0.0\\",
    \\"prettier\\": \\"^1.0.0\\"
  },
  \\"prettier\\": \\"@c4605/toolconfs/prettierrc\\",
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` install eslint-plugin-prettier when including prettier match file structure snapshot: file structure 1`] = `
Array [
  "/.eslintrc.json",
  "/package.json",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Eslint\` install eslint-plugin-prettier when including prettier match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Githooks\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"git-hook-pure\\": \\"^1.0.0\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Githooks\` match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Githooks\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Jest\` match file content snapshot: file content: /jest.config.js 1`] = `
"module.exports = {
  transform: {
    '^.+\\\\\\\\.js$': 'babel-jest',
    '^.+\\\\\\\\.ts$': 'ts-jest',
  },
  testMatch: [
    '**/?(*.)+(spec|test).ts',
  ],
  testPathIgnorePatterns: [
    '/node_modules/',
  ],
  moduleFileExtensions: [
    'ts',
    'js',
    'json',
    'node',
  ],
}
"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Jest\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"jest\\": \\"^1.0.0\\",
    \\"jest-cli\\": \\"^1.0.0\\",
    \\"ts-jest\\": \\"^1.0.0\\",
    \\"@types/jest\\": \\"^1.0.0\\"
  },
  \\"scripts\\": {
    \\"test\\": \\"jest ./src\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Jest\` match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
  "/jest.config.js",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Jest\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.LintStaged\` match file content snapshot: file content: /lint-staged.config.js 1`] = `
"module.exports = require('@c4605/toolconfs/lint-staged.config')
"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.LintStaged\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"lint-staged\\": \\"^1.0.0\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.LintStaged\` match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
  "/lint-staged.config.js",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.LintStaged\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Prettier\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"prettier\\": \\"^1.0.0\\"
  },
  \\"prettier\\": \\"@c4605/toolconfs/prettierrc\\",
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Prettier\` match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Prettier\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Renovate\` match file content snapshot: file content: /package.json 1`] = `
"{
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Renovate\` match file content snapshot: file content: /renovate.json 1`] = `
"{
  \\"extends\\": [\\"github>bolasblack/js-toolconfs:renovate\\"],
  \\"baseBranches\\": [\\"develop\\"]
}
"
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Renovate\` match file structure snapshot: file structure 1`] = `
Array [
  "/renovate.json",
  "/package.json",
]
`;

exports[`proj-tools with option \`include\` with \`IncludeItem.Renovate\` match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`include\` works with empty value match file structure snapshot: file structure 1`] = `Array []`;

exports[`proj-tools with option \`include\` works with empty value match tasks snapshot: rule tasks 1`] = `Array []`;

exports[`proj-tools with option \`jestReact\` do nothing if \`jest\` not included match file structure snapshot: file structure 1`] = `Array []`;

exports[`proj-tools with option \`jestReact\` do nothing if \`jest\` not included match tasks snapshot: rule tasks 1`] = `Array []`;

exports[`proj-tools with option \`jestReact\` in common case match file content snapshot: file content: /jest.config.js 1`] = `
"module.exports = {
  transform: {
    '^.+\\\\\\\\.jsx?$': 'babel-jest',
    '^.+\\\\\\\\.tsx?$': 'ts-jest',
  },
  snapshotSerializers: [
    'enzyme-to-json/serializer',
  ],
  moduleNameMapper: {
    '\\\\\\\\.(css|less|scss)$': 'identity-obj-proxy',
  },
  testMatch: [
    '**/?(*.)+(spec|test).ts?(x)',
  ],
  testPathIgnorePatterns: [
    '/node_modules/',
  ],
  moduleFileExtensions: [
    'ts',
    'tsx',
    'js',
    'jsx',
    'json',
    'node',
    'html',
  ],
}
"
`;

exports[`proj-tools with option \`jestReact\` in common case match file content snapshot: file content: /package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"jest\\": \\"^1.0.0\\",
    \\"jest-cli\\": \\"^1.0.0\\",
    \\"ts-jest\\": \\"^1.0.0\\",
    \\"@types/jest\\": \\"^1.0.0\\",
    \\"enzyme\\": \\"^1.0.0\\",
    \\"enzyme-to-json\\": \\"^1.0.0\\",
    \\"enzyme-adapter-react-16\\": \\"^1.0.0\\",
    \\"identity-obj-proxy\\": \\"^1.0.0\\"
  },
  \\"scripts\\": {
    \\"test\\": \\"jest ./src\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`jestReact\` in common case match file structure snapshot: file structure 1`] = `
Array [
  "/package.json",
  "/jest.config.js",
]
`;

exports[`proj-tools with option \`jestReact\` in common case match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": undefined,
      },
    },
  },
]
`;

exports[`proj-tools with option \`jestReact\` support \`cwd\` option match file content snapshot: file content: /test/jest.config.js 1`] = `
"module.exports = {
  transform: {
    '^.+\\\\\\\\.js$': 'babel-jest',
    '^.+\\\\\\\\.ts$': 'ts-jest',
  },
  testMatch: [
    '**/?(*.)+(spec|test).ts',
  ],
  testPathIgnorePatterns: [
    '/node_modules/',
  ],
  moduleFileExtensions: [
    'ts',
    'js',
    'json',
    'node',
  ],
}
"
`;

exports[`proj-tools with option \`jestReact\` support \`cwd\` option match file content snapshot: file content: /test/package.json 1`] = `
"{
  \\"devDependencies\\": {
    \\"jest\\": \\"^1.0.0\\",
    \\"jest-cli\\": \\"^1.0.0\\",
    \\"ts-jest\\": \\"^1.0.0\\",
    \\"@types/jest\\": \\"^1.0.0\\"
  },
  \\"scripts\\": {
    \\"test\\": \\"jest ./src\\"
  },
  \\"dependencies\\": {
    \\"@c4605/toolconfs\\": \\"^1.0.0\\"
  }
}"
`;

exports[`proj-tools with option \`jestReact\` support \`cwd\` option match file structure snapshot: file structure 1`] = `
Array [
  "/test/package.json",
  "/test/jest.config.js",
]
`;

exports[`proj-tools with option \`jestReact\` support \`cwd\` option match tasks snapshot: rule tasks 1`] = `
Array [
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": "test",
      },
    },
  },
  Object {
    "name": "@c4g/shell-exec-task",
    "options": Object {
      "args": Array [],
      "command": "yarn",
      "options": Object {
        "cwd": "test",
      },
    },
  },
]
`;
